system/security/cryptohelper/cryptocommon.cpp

```diff
@@ -0,0 +1,85 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#if defined(_USE_OPENSSL) && !defined(_WIN32)
+
+#include "jliball.hpp"
+#include <openssl/pem.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include "cryptocommon.hpp"
+
+
+static void addAlgorithms()
+{
+    OpenSSL_add_all_algorithms();
+}
+
+MODULE_INIT(INIT_PRIORITY_STANDARD)
+{
+    addAlgorithms();
+    return true;
+}
+MODULE_EXIT()
+{
+}
+
+
+namespace cryptohelper
+{
+
+IException *makeEVPExceptionVA(int code, const char *format, va_list args)
+{
+    StringBuffer formatMessage;
+    char *evpErr = formatMessage.reserve(120);
+    ERR_error_string_n(ERR_get_error(), evpErr, 120);
```

Comment:
Actually I copied the 120 from your digisign code:
```
#define EVP_THROW(str) {                                            \
                         char buff[120];                            \
                         ERR_error_string(ERR_get_error(), buff);   \
                         throw MakeStringException(-1, str, buff);  \
                       }
```

and in fact that calls ERR_error_string with can overrun the end of the buffer.
I probably assumed 120 was there because it was a known max(?)
I can't see a known max, but I'll up it.

And I'll fix EVP_THROW in this PR too to avoid the possible buffer run there.